lib/dor/text_extraction/cocina_updater.rb

```diff
@@ -13,13 +13,14 @@ module TextExtraction
     # 2. If the file will overwrite an existing file in Cocina ensure that
     #    the file to be overwritten was generated by SDR and has not been
     #    corrected.
-    # 3. If the file already exists in the Cocina update it by first deleting
+    # 3. If the file already exists in the Cocina, update it by first deleting
     #    from the Cocina and then adding it again.
-    # 4. If the file does not exist in the Cocina already look find a resource
-    #    that contains the same filename "stem" and add the file there. e.g.
-    #    "abc123.xml" would be added to the same resource that contains
-    #    "abc123.tiff".
-    #
+    # 4. If the file does not exist in the Cocina already, look for a resource
+    #    that contains the same filename "stem" or the same filename with extension added to the end of the stem
+    #    and add the file there.
+    #    e.g. "abc123.xml" would be added to the same resource that contains "abc123.tiff"
+    #    and "abc123_tiff.xml" would be added to the same resource that contains "abc123.tiff"
+    #    see https://github.com/sul-dlss/common-accessioning/issues/1443 for the logic behind this
```

Comment:
This makes it sound like the renaming will happen for OCR as well as Speech to Text jobs? I thought from conversation in Slack it was only the latter?

Reply:
I can clarify in the code comments.  There are two parts to this:

1. Renaming of files by adding the file name extension into the filename before sending them off to be processed.  This is currently only done only for captioning
2. When updating cocina at the end, the bundling of output files into the appropriate resource, either by matching filenames exactly or by using our new "includes the file extension" rule.  Either way is supported ensuring backwards compatibility.  So non-renamed OCR files will work and renamed caption files will work.  If we were to also rename OCR files before sending them off to ABBYY, this would also work.

This code comment is the cocina updater class, hence why it apples to any scenario.

Reply:
I added a very small tweak to hopefully make it more clear, but i think the code comment explains what the code is actually doing already, and links to the ticket